src/registry/routes/component.js

```diff
@@ -1,11 +1,32 @@
 'use strict';
 
 var GetComponentHelper = require('./helpers/get-component');
+var _ = require('underscore');
 
 module.exports = function(conf, repository){
 
   var getComponent = new GetComponentHelper(conf, repository);
 
+  var setCustomHeaders = function(req, res, componentResponse) {
+    if (req.params.componentVersion === componentResponse.version) {
+      //strong version request
+      res.set(componentResponse.headers);
+    } else {
+      //weak version request, therefore skip the blacklisted headers if any
+      if (_.isEmpty(res.conf.customHeadersToSkipOnWeakVersion || [])) {
+          res.set(componentResponse.headers);
+      } else {
+        var headersToSkip = new Set(res.conf.customHeadersToSkipOnWeakVersion);
+
+        _.forEach(Object.keys(componentResponse.headers), function(header) {
+          if (!headersToSkip.has(header.toLowerCase())) {
```

Comment:
I was thinking that if we change [setHeader to sanitise the headers by "lowercasing" it](https://github.com/opentable/oc/blob/74792a8ecfa6c0cf04d0a4ce9d5eeecabe48a039/src/registry/routes/helpers/get-component.js#L249) then we may not need to "lowercase" it here?